artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContext.java

```diff
@@ -95,7 +100,11 @@ public void initialize() throws Exception {
             defRoutingType = getRoutingType(target.getCapabilities(), address);
 
             try {
-               sessionSPI.createTemporaryQueue(address, defRoutingType);
+               if (defRoutingType == RoutingType.ANYCAST) {
+                  sessionSPI.createTemporaryQueue(address, defRoutingType);
```

Comment:
It doesnt seem like the address generated just above this changed either, though I notice it being manipulated elsewhere to add and remove the namespace. Wouldnt it be simpler to have the broker-generated address actually include the namespace its considered to be part of? Then the address used everywhere is simply the same actual queue/address name.